framework/include/base/MooseApp.h

```diff
@@ -259,14 +259,14 @@ class MooseApp : public ConsoleStreamInterface, public libMesh::ParallelObject
   void setExecutioner(std::shared_ptr<Executioner> && executioner) { _executioner = executioner; }
 
   /**
-   * Set a Boolean indicating whether this app will use a Nonlinear or Eigen System.
+   * Get the problem type for this App
    */
-  bool & useNonlinear() { return _use_nonlinear; }
+  const std::string & problemType() const { return _problem_type; }
 
   /**
-   * Set a Boolean indicating whether this app will use an eigenvalue executioner.
+   * Set the problem type for this App
```

Comment:
"Returns a writable reference to the problem type or this App."

Real setters do things like: `void setter(TYPE a) { _member_variable = a; }`, so the doco is a bit misleading.